robottelo/api/utils.py

```diff
@@ -437,3 +438,34 @@ def configure_provisioning(org=None, loc=None):
         'host_group': host_group.name,
         'domain': domain.name,
     }
+
+
+def get_role_by_bz(bz_id):
+    """Create and configure custom role entity for the testing of specific bugs
+     This function will read the dictionary of permissions and their associated
+     bugzilla id's from robottelo.constants "PERMISSIONS_WITH_BZ",
+     these permissions will create filter and a single role will be created
+     from all the filters.
+
+     :param bz_id: This is the bugzilla id that is specified in the
+        PERMISSIONS_WITH_BZ list, all the permissions associated with the bz_id
+        will be fetched and filters will be created
+     :return: A single role entity will be created from all the created filters
+     """
+    perms_dict = {}
+    role = entities.Role().create()
+    for res_type in PERMISSIONS_WITH_BZ.keys():
+        for value in PERMISSIONS_WITH_BZ[res_type]:
+            if bz_id in value['bz']:
```

Comment:
For that case I prefer to have it in dict set as None

```python
{'name': 'app_root', 'bz': None}
```

and then in code to keep the iteration syntax clean I always do:

```python
if bz_id in value.get('bz', []):
   # do something
```

So if 'bz' is None it always get an empty list without he need of external lib, and it is short circuit, no overhead at all as empty list will not be iterated.

@rplevka @renzon